File: misago/emberapp/app/models/username-change.js
import DS from 'ember-data';

export default DS.Model.extend({
  user: DS.belongsTo('User'),
  changed_by: DS.belongsTo('User'),
  changed_by_username: DS.attr('string'),
  changed_by_slug: DS.attr('string'),
  changed_on: DS.attr('moment-date'),
  new_username: DS.attr('string'),
  old_username: DS.attr('string'),

  intId: function() {
    return + this.get('id');
  }.property('id')
});

File: misago/users/serializers/usernamechange.py
from rest_framework import serializers

from misago.users.models import UsernameChange

from .user import UserSerializer as BaseUserSerializer


__all__ = ['UsernameChangeSerializer']

UserSerializer = BaseUserSerializer.subset_fields('id', 'username', 'avatars', 'url')


class UsernameChangeSerializer(serializers.ModelSerializer):
    user = UserSerializer(many=False, read_only=True)
    changed_by = UserSerializer(many=False, read_only=True)

    class Meta:
        model = UsernameChange
        fields = [
            'id',
            'user',
            'changed_by',
            'changed_by_username',
            'changed_on',
            'new_username',
            'old_username',
        ]
